src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1559,7 +1566,7 @@ private static void AddMemberLookupSymbolsInfoInNamespace(LookupSymbolsInfo resu
         {
             foreach (var symbol in GetCandidateMembers(ns, options, originalBinder))
             {
-                if (originalBinder.CanAddLookupSymbolInfo(symbol, options, null))
+                if (originalBinder.CanAddLookupSymbolInfo(symbol, options, result, null))
```

Comment:
Why are we filtering symbols after lookup, instead of using the overload `GetCandidateMembers(NamespaceOrTypeSymbol nsOrType, string name, LookupOptions options, Binder originalBinder)` (with an additional `string name` parameter)?

It seems like we could get a bit better perf to not have to allocate a large `ImmutableArray` that holds all results, and only then cut out the items that are named incorrectly.